File: T1.SqlSharp/T1.SqlSharp/Expressions/SqlInnerTableSource.cs
using T1.Standard.IO;

namespace T1.SqlSharp.Expressions;

public class SqlInnerTableSource : SqlTableSource
{
    public new SqlType SqlType { get; } = SqlType.InnerTableSource;
    public required ISqlExpression Inner { get; set; }
    
    public override string ToSql()
    {
        var sql = new IndentStringBuilder();
        sql.Write(Inner.ToSql());
        if (!string.IsNullOrEmpty(Alias))
        {
            sql.Write($" AS {Alias}");
        }
        if (Withs.Count > 0)
        {
            sql.Write(" WITH(");
            foreach (var with in Withs.Select((value,index)=> new {value, index}))
            {
                sql.Write($"{with.value.ToSql()}");
                if (with.index < Withs.Count - 1)
                {
                    sql.Write(", ");
                }
            }
            sql.Write(")");
        }
        return TableName;
    }
}

File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlInnerTableSource.ts
import { SqlType } from './SqlType';
import { TextSpan } from '../StringParser';
import { ISqlExpression, SqlVisitor } from './ISqlExpression';
import { SqlTableSource } from './SqlTableSource';

export class SqlInnerTableSource extends SqlTableSource {
    SqlType: SqlType = SqlType.InnerTableSource;
    Inner!: ISqlExpression;

    Accept(visitor: SqlVisitor): void {
        visitor.Visit_InnerTableSource(this);
    }

    ToSql(): string {
        let sql = this.Inner.ToSql();
        // 假設 Alias 與 Withs 屬性繼承自 SqlTableSource
        // @ts-ignore
        if (this.Alias) {
            sql += ` AS ${this.Alias}`;
        }
        // @ts-ignore
        if (this.Withs && this.Withs.length > 0) {
            sql += ' WITH(' + this.Withs.map((w: ISqlExpression) => w.ToSql()).join(', ') + ')';
        }
        return sql;
    }
} 